Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -98,20 +102,15 @@ final class ConfigGenerator: ConfigGenerating {
                                             fileElements: ProjectFileElements,
                                             pbxproj: PBXProj,
                                             configurationList: XCConfigurationList) throws {
-        let variant: BuildSettingsProvider.Variant = (buildConfiguration.variant == .debug) ? .debug : .release
-        let defaultConfigSettings = BuildSettingsProvider.projectDefault(variant: variant)
-        let defaultSettingsAll = BuildSettingsProvider.projectDefault(variant: .all)
-
-        var settings: [String: Any] = [:]
-        extend(buildSettings: &settings, with: defaultSettingsAll)
-        extend(buildSettings: &settings, with: project.settings.base)
-        extend(buildSettings: &settings, with: defaultConfigSettings)
+        var settings = defaultSettingsProvider.defaultProjectSettings(project: project,
+                                                                      buildConfiguration: buildConfiguration)
+        settingsHelper.extend(buildSettings: &settings, with: project.settings.base)
```

Comment:
You are holding in an instance variable, `settingsHelper`, a state that is tied to the generate method. I'd rather create the `SettingsHelper` instance in the method body.